packages/@aws-cdk/lambda-layer-awscli/package.json

```diff
@@ -71,25 +71,32 @@
     "organization": true
   },
   "license": "Apache-2.0",
+  "bundledDependencies": [
+    "semver"
```

Comment:
I'm not sure I did this correctly... Adding the non-JSII package as a dependency forced me to add it here as well. And then having a bundled dependency led me to add it to the "nohoist" list in our top-level package.json using `yarn build --fix`. It's strange because `semver` is already a dependency of `aws-cdk-lib`, so this will get ignored in the release anyway.

Reply:
Potentially, we could add a utility function in `@aws-cdk/core` that does the version range check and uses the `semver` bundled there. Just available to TS/JS, nothing else would be good enough to start. Something like:

```ts
export function versionMatchesRange(version: string, range: string) {
  // Use semver here
}
```